### ISSUE TITLE
[Feature Request]: Use Fast Util's map for the claimedChunks field in ClaimedChunkManager

### ISSUE THREAD

**Issue by Kasualix** [FTBTeam/FTB-Mods-Issues#980]
### Mod

FTB Chunks

### Description

Traditional HashMap is causing a performance issue.
[Spark Profile](https://spark.lucko.me/fAOJIkykxx) (_[Backup](https://github.com/FTBTeam/FTB-Mods-Issues/files/13060874/Profile.zip)_)
[Related Codes](https://github.com/FTBTeam/FTB-Chunks/blob/b4897b12b5d0091d670581fede7ad353acfe01e6/common/src/main/java/dev/ftb/mods/ftbchunks/data/ClaimedChunkManagerImpl.java#L50)
DoesPotatoTick is using FTB Chunks' API. You can see the lag in mods->doespotatotick.

### Describe the solution you'd like

Don't use Traditional HashMap.
Use Object2ObjectOpenHashMap instead (or maybe Reference2ReferenceOpenHashMap? or Reference2Object/Object2Reference? I don't know much about them).
Anyway if you change it I can remove my [mixin](https://github.com/MCTeamPotato/DoesPotatoTick/blob/Fabric/quilt-1201/src/main/java/com/teampotato/doespotatotick/mixin/ClaimedChunkManagerImplMixin.java).

**Comment by desht** [comment]
A few comments here:

* This is 1.16.5 so changes are not likely here. If you have a mixin that works for you, I would say stick with that. Since no changes are likely for the 1.16.5 codebase, your mixin probably won't break.
* Have you done any empirical testing on the performance of `HashMap` vs `Object2ObjectOpenHashMap` for this key type?
* Looking at the Spark profile, it appears the majority of time is spent in getting the hashcode for `ChunkDimPos`, which wouldn't be affected by changing the map implementation. Or am I missing something in the profile?
* Recent versions of FTB Library use a `ChunkDimPos` implementation which caches the hashcode on construction (since it's an immutable object), meaning hash code retrieval during usage is very fast. So I doubt we would see similar performance issues in 1.19/1.20.

**Comment by Kasualix** [comment]
Yes I have tested. With Object2ObjectOpenHashMap, even if many chunks in my server are claimed, severe performance issues about it no longer occur.

**Comment by Kasualix** [comment]
I will remove the ClaimedChunkManagerImplMixin for 1.19.2+ in the next version of DPT, thanks for your info.

**Comment by desht** [comment]
I can't guarantee it will work, and I will seriously consider switching to a fastutils map here. Depending on how & when the ChunkDimPos keys are constructed, caching the hash at construction time won't necessarily help.

So gonna leave this open for now.
